src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+    }
+
+    @Test
+    void userTooOld_ExceptionMessage_Ok() {
+        User userTooOld = createUser(100, "crimeabridgedestroyed", "Himars");
+        try {
+            registrationService.register(userTooOld);
+        } catch (RuntimeException e) {
+            String actualMessage = e.getMessage();
+            String expectedMessage = "User too old!";
+            assertEquals(actualMessage, expectedMessage);
+        }
+    }
+
+    @Test
+    void userNegativeAge_ExceptionMessage_Ok() {
+        User userNegativeAge = createUser(-100, "crimeabridgedestroyed", "Himars");
+        try {
+            registrationService.register(userNegativeAge);
+        } catch (RuntimeException e) {
+            String actualMessage = e.getMessage();
+            String expectedMessage = "User's age cannot be negative!";
+            assertEquals(actualMessage, expectedMessage);
+        }
+    }
+
+    @Test
+    void userNegativeAge_Exception_Ok() {
+        User userNegativeAge = createUser(-100, "crimeabridgedestroyed", "Himars");
+        assertThrows(RuntimeException.class, () -> registrationService.register(userNegativeAge));
+    }
+
+    @Test
+    void userNullAge_Exception_Ok() {
+        User userNullAge = createUser(null, "crimeabridgedestroyed", "Himars");
+        assertThrows(NullPointerException.class, () -> registrationService.register(userNullAge));
+    }
+
+    @Test
+    void userNullAge_ExceptionMessage_Ok() {
+        User userNullAge = createUser(null, "crimeabridgedestroyed", "Himars");
+        try {
+            registrationService.register(userNullAge);
+        } catch (NullPointerException e) {
+            String actualMessage = e.getMessage();
+            String expectedMessage = "User's age should not be null";
+            assertEquals(actualMessage, expectedMessage);
+        }
+    }
+
+    @Test
+    void userNullPassword_Exception_Ok() {
+        User userNullPassword = createUser(80, null, "Himars");
+        assertThrows(NullPointerException.class,
+                () -> registrationService.register(userNullPassword));
+    }
+
+    @Test
+    void userNullPassword_ExceptionMessage_Ok() {
+        User userNullPassword = createUser(80, null, "Himars");
+        try {
+            registrationService.register(userNullPassword);
+        } catch (NullPointerException e) {
+            String actualMessage = e.getMessage();
+            String expectedMessage = "Password should not be null";
+            assertEquals(actualMessage, expectedMessage);
+        }
+    }
+
+    @Test
+    void userNullLogin_Exception_Ok() {
+        User userNullLogin = createUser(80, "crimeabridgedestroyed", null);
+        assertThrows(NullPointerException.class, () -> registrationService.register(userNullLogin));
+    }
+
+    @Test
+    void userNullLogin_ExceptionMessage_Ok() {
+        User userNullLogin = createUser(80, "crimeabridgedestroyed", null);
```

Comment:
we can place valid user creation in the setUp block, for example
   @BeforeEach
    void setUp() {
        validUser = new User();
        validUser.setLogin(DEFAULT_LOGIN);
        validUser.setPassword(DEFAULT_PASSWORD);
        validUser.setAge(DEFAULT_AGE);
    }

Reply:
I don't think I should do it because I don't need a valid user before each test. I have many tests that check invalid fields, and creating a valid user each time will waste resources